state/pgstatestorage.go

```diff
@@ -949,6 +950,21 @@ func (p *PostgresStorage) AddBatchNumberInForcedBatch(ctx context.Context, force
 	return err
 }
 
+// GetBatchNumberOfL2Block gets a batch number for l2 block by its number
+func (p *PostgresStorage) GetBatchNumberOfL2Block(ctx context.Context, blockNumber uint64, dbTx pgx.Tx) (uint64, error) {
```

Comment:
It would be nice to test this function, since in the jsonrpc this is mocked

Reply:
Done please review again. :)